IOPool/Streamer/src/StreamerInputFile.cc

```diff
@@ -111,16 +119,48 @@ namespace edm {
     currentFileOpen_ = false;
   }
 
-  storage::IOSize StreamerInputFile::readBytes(char* buf, storage::IOSize nBytes) {
+  std::pair<storage::IOSize, char*> StreamerInputFile::readBytes(char* buf,
+                                                                 storage::IOSize nBytes,
+                                                                 bool zeroCopy,
+                                                                 unsigned int skippedHdr) {
     storage::IOSize n = 0;
+    char* ptr = buf - skippedHdr;
```

Comment:
After spending some time in wondering why `ptr` is initialized to `buf - skippedHdr` I'd suggest to add a comment that in case of successful read (`n == nBytes`), the caller expects the returned pointer to point to the beginning of the header.

Reply:
Added:
```
    //returned pointer should point to the beginning of the header
    //even if we read event payload that comes afterwards
```
and a couple of more comments in that function.